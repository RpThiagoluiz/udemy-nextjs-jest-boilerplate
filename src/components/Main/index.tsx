import * as S from './styles'
//Assim, de olhar vc ja sabe q isso Ã© um component de stylo.

const Main = () => (
  <S.Wrapper>
    <S.Logo
      src="/img/logo.svg"
      alt="Imagem de um atamo e React Avancando escrito ao lado"
    />
    <S.Title>React Avancado</S.Title>
    <S.Description>
      TypeScript, ReactJS, NextJS e Styled Components
    </S.Description>
    <S.Ilustration
      src="/img/hero-illustration.svg"
      alt="Um desenvolvedor de frente para uma tela com codigo."
    />
  </S.Wrapper>
)

export default Main
