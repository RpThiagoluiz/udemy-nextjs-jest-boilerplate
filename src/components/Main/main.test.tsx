import { render, screen } from '@testing-library/react'
import Main from '.'
describe('<Main/>', () => {
  it('should render the heading', () => {
    render(<Main />)

    expect(
      screen.getByRole('heading', { name: /react avancado/i })
    ).toBeInTheDocument()
  })

  it('should be render in Snapshot ', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /react avancado/i })
    ).toBeInTheDocument()

    //Criar uma pasta e dentro dela vai ter o snapshot dela. que vai verificar se ta tudo igual. Caso vc mude alguma coisa dentro ele vai avisar que vc mudou algo dentro do codigo, e vc consegue reverter e\ou autorizar isso msm.
    //Garante uma camada a mais de seguranca - como estamos procurando um reading precisamos dele.
    expect(container.firstChild).toMatchSnapshot()
  })
})
