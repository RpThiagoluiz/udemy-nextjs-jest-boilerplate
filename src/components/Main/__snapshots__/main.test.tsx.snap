// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main/> should be render in Snapshot  1`] = `
<main>
  <h1>
    React Avancado
  </h1>
</main>
`
